src/animation.js

```diff
@@ -130,13 +132,39 @@
         return 'pending';
       if (this.paused)
         return 'paused';
-      if (this.finished)
+      if (this._isFinished)
         return 'finished';
       return 'running';
     },
+    get finished() {
+      if (!this._finishedPromise) {
```

Comment:
Rather than this approach, would it be better to create the `_finishedPromise` initially, as [specified in 3.5.12](http://w3c.github.io/web-animations/#the-current-finished-promise)?

You have to check for transitions anyway-
* into 'finished', to resolve the `Promise`
* out of 'finished', to create a new `Promise`
but it looks like that might be tricky based on the way you return the current `playState`, since that's determined at the time.... thoughts? Maybe inside the tick methods?